apps/emqx_authz/src/emqx_authz_api_mnesia.erl

```diff
@@ -0,0 +1,556 @@
+                                },
+                                all => #{
+                                    summary => <<"All">>,
+                                    value => jsx:encode(?EXAMPLE_ALL)
+                                }
+                            }
+                        }
+                    }
+                },
+                <<"404">> => emqx_mgmt_util:bad_request(<<"Not Found">>)
+            }
+        },
+        put => #{
+            description => "Update one record",
+            parameters => [
+                #{
+                    name => type,
+                    in => path,
+                    schema => #{
+                       type => string,
+                       enum => [<<"username">>, <<"clientid">>]
+                    },
+                    required => true
+                },
+                #{
+                    name => key,
+                    in => path,
+                    schema => #{
+                       type => string
+                    },
+                    required => true
+                }
+            ],
+            requestBody => #{
+                content => #{
+                    'application/json' => #{
+                        schema => minirest:ref(<<"record">>),
+                        examples => #{
+                            username => #{
+                                summary => <<"Username">>,
+                                value => jsx:encode(?EXAMPLE_USERNAME)
+                            },
+                            clientid => #{
+                                summary => <<"Clientid">>,
+                                value => jsx:encode(?EXAMPLE_CLIENTID)
+                            }
+                        }
+                    }
+                }
+            },
+            responses => #{
+                <<"204">> => #{description => <<"Updated">>},
+                <<"400">> => emqx_mgmt_util:bad_request()
+            }
+        },
+        delete => #{
+            description => "Delete one record",
+            parameters => [
+                #{
+                    name => type,
+                    in => path,
+                    schema => #{
+                       type => string,
+                       enum => [<<"username">>, <<"clientid">>]
+                    },
+                    required => true
+                },
+                #{
+                    name => key,
+                    in => path,
+                    schema => #{
+                       type => string
+                    },
+                    required => true
+                }
+            ],
+            responses => #{
+                <<"204">> => #{description => <<"No Content">>},
+                <<"400">> => emqx_mgmt_util:bad_request()
+            }
+        }
+    },
+    {"/authorization/sources/built-in-database/:type/:key", Metadata, record}.
+
+purge(delete, _) ->
```

Comment:
ensure 'built-in-database' type source is disabled before purge.

Reply:
Done